pyccel/ast/operators.py

```diff
@@ -198,21 +198,21 @@ def __init__(self, arg):
         super().__init__(arg)
 
     @staticmethod
-    def _calculate_dtype(_args):
+    def _calculate_dtype(*_args):
         """ Sets the dtype and precision
         They are chosen to match the argument
         """
-        a = _args[0]
+        a = _args[0][0]
```

Comment:
Why `[0][0]`? Is this because you are calling this as:
```
self._calculate_dtype(self.args)
```
?
You should do:
```
self._calculate_dtype(*self.args)
```
This way `_args` and `self.args` will contain the same thing, and you can even do:
```
def _calculate_dtype(arg1, arg2):
```
in the classes where this makes sense

Reply:
I edited it in [6efbe03](https://github.com/pyccel/pyccel/pull/797/commits/6efbe034730311dd0100d5dd5d9f6444449e94f1)